app/src/ui/branches/ci-status.tsx

```diff
@@ -83,43 +84,87 @@ export class CIStatus extends React.PureComponent<
     this.unsubscribe()
   }
 
-  private onStatus = (status: IAPIRefStatus | null) => {
-    this.setState({ status })
+  private onStatus = (check: ICombinedRefCheck | null) => {
+    this.setState({ check })
   }
 
   public render() {
-    const { status } = this.state
+    const { check } = this.state
 
-    if (status === null || status.total_count === 0) {
+    if (check === null || check.checks.length === 0) {
       return null
     }
 
-    const title = getRefStatusSummary(status)
-    const state = status.state
-
     return (
       <Octicon
         className={classNames(
           'ci-status',
-          `ci-status-${state}`,
+          `ci-status-${getClassNameForCheck(check)}`,
           this.props.className
         )}
-        symbol={getSymbolForState(state)}
-        title={title}
+        symbol={getSymbolForCheck(check)}
+        title={getRefCheckSummary(check)}
       />
     )
   }
 }
 
-function getSymbolForState(state: APIRefState): OcticonSymbol {
-  switch (state) {
-    case 'pending':
-      return OcticonSymbol.primitiveDot
+function getSymbolForCheck(check: ICombinedRefCheck): OcticonSymbol {
+  switch (check.conclusion) {
+    case 'timed_out':
+      return OcticonSymbol.x
     case 'failure':
       return OcticonSymbol.x
+    case 'neutral':
+      return OcticonSymbol.primitiveSquare
     case 'success':
       return OcticonSymbol.check
-    default:
-      return assertNever(state, `Unknown state: ${state}`)
+    case 'cancelled':
+      return OcticonSymbol.stop
+    case 'action_required':
+      return OcticonSymbol.alert
+    case 'skipped':
+      return OcticonSymbol.skip
+    case 'stale':
+      return OcticonSymbol.issueReopened
```

Comment:
It's very nice to have specific icons and messages for all the potential conclusions, but if I understand the code correctly the combined conclusion currently can only be `success`, `failure` or `null`.

Is the plan to make the logic that calculates the combined state a bit smarter to make it more granular? (I think this can be done in a separate PR)

Reply:
Whoopsie, great catch! The intention here was to support all the different conclusions when there's only a single check but I forgot about that along the way. Change incoming momentarily. A medium term goal for us is to allow peeking into a combined status to see exactly what's going on similar to how dotcom does it which is why the checks array is kept in the combined check to begin with.

> Is the plan to make the logic that calculates the combined state a bit smarter to make it more granular? (I think this can be done in a separate PR)

Same thing here as with the incomplete/failure distinction. The logic is mirroring that of how dotcom does check rollups AFAICT. There's room for improvement but if we change it here we should do so on the dotcom side as well.

Reply:
Thanks for the context! Makes sense!